docs: explanation on using protocols with partitions (#17759)
electron/electron: docs/api/protocol.md

**Note:** All methods unless specified can only be used after the `ready` event
of the `app` module gets emitted.

> ## Using `protocol` with a `partition` in `webPreferences`

Reviewer: ```suggestion
## Using `protocol` with a custom `partition` or `session`
```

> A protocol is registered to the default Electron session. If you define a `partition` on your `browserWindow`'s `webPreferences`, then that partition will use a different session and your custom protocol will not work.

Reviewer: ```suggestion
A protocol is registered to a specific Electron [`session`](./session.md) object. If you define a `partition` or `session` on your `browserWindow`'s `webPreferences`, then that window will use a different session and your custom protocol will not work if you just use `electron.protocol.XXX`.
```

> To have your custom protocol work in combination with a partition, you need to register it to the session of that partition:

Reviewer: ```suggestion
To have your custom protocol work in combination with a custom session, you need to register it to that session explicitly.
```

```javascript
const { session, app, protocol } = require('electron')
const path = require('path')

app.on('ready', () => {
  const partition = 'persist:example'
  const ses = session.fromPartition(partition)

  ses.protocol.registerFileProtocol('atom', (request, callback) => {
    const url = request.url.substr(7)
    callback({ path: path.normalize(`${__dirname}/${url}`) })
  }, (error) => {
    if (error) console.error('Failed to register protocol')
  })

  mainWindow = new BrowserWindow({
    width: 800,
    height: 600,
    webPreferences: {
      partition: partition
    }
  })
})
```

> Using `protocol.registerStandardSchemes` without the session will still register your custom protocol as a standard scheme.

Reviewer: Although not something that needs to be fixed here, this is confusing / unexplainable.  We should separate those APIs if some are per-session and others aren't.

Author: There's also the (currently undocumented) getStandardSchemes, which behaves the same.
